### PR TITLE
Sw dal events venue linking

### PR DESCRIPTION
**Pull request by SWvheerden**
DAL and unit tests done for events, organisation and venue linking

### CODE REVIEW

File: src/models/events.rs [hunk 1]
```diff
@@ -66,6 +66,52 @@ impl Event {
             events::table.find(id).first::<Event>(conn.get_connection()),
         )
     }
+    pub fn find_via_venue(venue_id: &Uuid, conn: &Connectable) -> Result<Event, DatabaseError> {
```
Comment (CjS77) [inline src/models/events.rs]: This doesn't look right. Returning the first event given a venue id doesn't sound terribly useful.
Further, if there's no event associated with a venue, I guess `first` will return an error, in which case, "Error loading event" is misleading.

File: src/models/events.rs [hunk 2]
```diff
@@ -66,6 +66,52 @@ impl Event {
...
+        DatabaseError::wrap(
+            ErrorCode::QueryError,
+            "Error loading event via venue",
+            events::table
+                .filter(events::venue_id.eq(venue_id))
+                .first::<Event>(conn.get_connection()),
+        )
+    }
+    pub fn find_via_organization(
```
Comment (CjS77) [inline src/models/events.rs]: Same comment as above

File: src/models/events.rs [hunk 3]
```diff
@@ -66,6 +66,52 @@ impl Event {
...
+    ) -> Result<Event, DatabaseError> {
+        DatabaseError::wrap(
+            ErrorCode::QueryError,
+            "Error loading events via organization",
+            events::table
+                .filter(events::organization_id.eq(organization_id))
+                .first::<Event>(conn.get_connection()),
+        )
+    }
+    pub fn find_via_venue_all(
```
Comment (CjS77) [inline src/models/events.rs]: This is better. I'm not sure we need the methods above.

Thought: would `find_all_in_venue|organization` be a better method name?

File: src/models/organization_venues.rs [hunk 4]
```diff
@@ -43,4 +44,77 @@ impl OrganizationVenue {
...
+    pub fn find(id: &Uuid, conn: &Connectable) -> Result<OrganizationVenue, DatabaseError> {
+        DatabaseError::wrap(
+            ErrorCode::QueryError,
+            "Error loading organization venue link",
+            organization_venues::table
+                .find(id)
+                .first::<OrganizationVenue>(conn.get_connection()),
+        )
+    }
+    pub fn find_via_venue(
```
Comment (CjS77) [inline src/models/organization_venues.rs]: Same comment as above

File: src/models/organization_venues.rs [hunk 5]
```diff
@@ -43,4 +44,77 @@ impl OrganizationVenue {
...
+    ) -> Result<OrganizationVenue, DatabaseError> {
+        DatabaseError::wrap(
+            ErrorCode::QueryError,
+            "Error loading organization venue link via venue",
+            organization_venues::table
+                .filter(organization_venues::venue_id.eq(venue_id))
+                .first::<OrganizationVenue>(conn.get_connection()),
+        )
+    }
+    pub fn find_via_organization(
```
Comment (CjS77) [inline src/models/organization_venues.rs]: Same comment as above

File: src/models/venues.rs [hunk 6]
```diff
@@ -67,4 +68,57 @@ impl Venue {
...
+        conn: &Connectable,
+    ) -> Result<Venue, DatabaseError> {
+        let venue_id = OrganizationVenue::find_via_organization(&organization_id, conn).unwrap();
+        DatabaseError::wrap(
+            ErrorCode::QueryError,
+            "Error loading organization venue link via organization",
+            Venue::find(&venue_id.venue_id, conn),
+        )
+    }
+    pub fn find_via_organization_all(
```
Comment (CjS77) [inline src/models/venues.rs]: maybe `find_all_for_organization`?

File: src/models/venues.rs [hunk 7]
```diff
@@ -67,4 +68,57 @@ impl Venue {
...
+        organization_id: &Uuid,
+        conn: &Connectable,
+    ) -> Result<Vec<Venue>, DatabaseError> {
+        let all_linked_organization_ids =
```
Comment (CjS77) [inline src/models/venues.rs]: I think you can simplify this by doing something like:

```
let venue_ids = OrganizationVenue::belonging_to(organization_id).select(organization_venues::venue_id);
    venues::table
        .filter(venues::id.eq(any(venue_ids)))
        .load::<Venue>(conn)
        ....
```

File: tests/unit/organization_venues.rs [hunk 8]
```diff
@@ -19,3 +19,124 @@ fn create() {
...
+fn find() {
+    //create user and organization
+    let project = TestProject::new();
+    let user = User::create("Jeff", "jeff@tari.com", "555-555-5555", "examplePassword")
+        .commit(&project)
+        .unwrap();
+    let mut edited_organization = Organization::create(user.id, "OrganizationForFindTest2")
+        .commit(&project)
+        .unwrap();
+    edited_organization.address = Some(<String>::from("Test Address2"));
```
Comment (CjS77) [inline tests/unit/organization_venues.rs]: Nit.. Do you need the turbofish notation here? Doesn't String::from("test address") work just fine?
you can also try
`"test address".into()` or
`"test address".to_string()`

matter of preference

File: src/models/venues.rs (hunk 7 above)
Comment (SWvheerden) [reply src/models/venues.rs]: I have leaved this for now with a todo, as wel as #26

**Pull request merged**